java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -136,47 +134,50 @@
  */
 JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OnnxTensor_createStringTensor
         (JNIEnv * jniEnv, jclass jobj, jlong apiHandle, jlong allocatorHandle, jobjectArray stringArr, jlongArray shape) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+    (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
     const OrtApi* api = (const OrtApi*) apiHandle;
     OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
 
     // Extract the shape information
     jboolean mkCopy;
-    jlong* shapeArr = (*jniEnv)->GetLongArrayElements(jniEnv,shape,&mkCopy);
-    jsize shapeLen = (*jniEnv)->GetArrayLength(jniEnv,shape);
+    jlong* shapeArr = (*jniEnv)->GetLongArrayElements(jniEnv, shape, &mkCopy);
+    jsize shapeLen = (*jniEnv)->GetArrayLength(jniEnv, shape);
 
     // Array length
     jsize length = (*jniEnv)->GetArrayLength(jniEnv, stringArr);
 
     // Create the OrtValue
-    OrtValue* ortValue;
-    checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator,(int64_t*)shapeArr,shapeLen,ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING,&ortValue));
-    (*jniEnv)->ReleaseLongArrayElements(jniEnv,shape,shapeArr,JNI_ABORT);
-
-    // Create the buffers for the Java strings
-    const char** strings;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(char*)*length,(void**)&strings));
-    jobject* javaStrings;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(jobject)*length,(void**)&javaStrings));
-
-    // Copy the java strings into the buffers
-    for (jsize i = 0; i < length; i++) {
-        javaStrings[i] = (*jniEnv)->GetObjectArrayElement(jniEnv,stringArr,i);
-        strings[i] = (*jniEnv)->GetStringUTFChars(jniEnv,javaStrings[i],NULL);
+    OrtValue* ortValue = NULL;
+    OrtErrorCode code = checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator, (int64_t*)shapeArr, shapeLen,
+                                        ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING, &ortValue));
+    (*jniEnv)->ReleaseLongArrayElements(jniEnv, shape, shapeArr, JNI_ABORT);
+
+    if (code == ORT_OK) {
+      // Create the buffers for the Java strings
+      const char** strings;
+      code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(char*) * length, (void**)&strings));
+
+      if (code == ORT_OK) {
+        // Copy the java strings into the buffers
+        for (jsize i = 0; i < length; i++) {
+            jobject javaString = (*jniEnv)->GetObjectArrayElement(jniEnv, stringArr, i);
+            strings[i] = (*jniEnv)->GetStringUTFChars(jniEnv, javaString, NULL);
+        }
+
+        // Assign the strings into the Tensor
+        checkOrtStatus(jniEnv, api, api->FillStringTensor(ortValue, strings, length));
```

Comment:
>checkOrtStatus [](http://example.com/codeflow?start=8&length=14)

What if this has an error, what happens if the next checkOrtStatus also errors out? Nested exceptions or just the last one?
Should we create a chain of exceptions?

Reply:
It returns the first exception thrown rather than any later ones.